batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/controller/StudentControllerServlet.java

```diff
@@ -0,0 +1,179 @@
+            create(request,response);
+        }
+        else if(pathParts.length == 3 && pathParts[2].equals("edit"))
+        {
+            if(pathParts[1].matches("[-+]?\\d*\\.?\\d+"))
+            {
+                System.out.println("center in a number");
+                edit(request, response, Integer.parseInt(pathParts[1]));
+            }
+            else{
+                System.out.println("!!!center in a not number");
+            }
+
+        }
+    }
+
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+        System.out.println("fetch the data of students");
+        List<Student> stdList = new ArrayList<Student>();
+        try {
+            int LIMIT = 20;
+            StudentService studentService = new StudentService();
+            stdList = studentService.fetch(page,LIMIT);
+            int totalCount = studentService.studentCount();
+            request.setAttribute("studentList", stdList);
+            request.setAttribute("page",page);
+            request.setAttribute("totalPages",totalCount/LIMIT);
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/display.jsp").forward(request, response);
+        } catch (DataException ex) {
+            System.out.println("--------------" + ex);
+        }
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/newStudent.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Student s=new Student();
+        s.setFirstName(request.getParameter("fname"));
+        s.setMiddleName(request.getParameter("mname"));
+        s.setLastName(request.getParameter("lname"));
+        s.setAddress(request.getParameter("address"));
+        s.setGrade(Integer.parseInt(request.getParameter("grade")));
+        try {
+            StudentService studentService = new StudentService();
+            studentService.save(s);
+            response.sendRedirect(request.getContextPath());
+        }catch (DataException ex) {
+            System.out.println("--------------" + ex);
+        }
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException {
+        try {
+            StudentService studentService = new StudentService();
+            Student s = studentService.fetchById(id);
+            request.setAttribute("student",s);
+            System.out.println(s.getAddress());
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/edit.jsp").forward(request, response);
+        }catch (DataException ex) {
+            System.out.println("--------------" + ex);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)